atc/db/versions_db.go

```diff
@@ -32,6 +32,27 @@ func NewVersionsDB(conn Conn, limitRows int, cache *gocache.Cache) VersionsDB {
 	}
 }
 
+func (versions VersionsDB) IsFirstOccurrence(ctx context.Context, jobID int, inputName string, versionMD5 ResourceVersion) (bool, error) {
+	var exists bool
+	err := versions.conn.QueryRowContext(ctx, `
+		WITH builds_of_job AS (
+			SELECT id FROM builds WHERE job_id = $1
+		)
+		SELECT EXISTS (
+			SELECT 1
+			FROM build_resource_config_version_inputs i
+			JOIN builds_of_job b ON b.id = i.build_id
+			WHERE i.name = $2
+			AND i.version_md5 = $3
+		)`, jobID, inputName, versionMD5).
```

Comment:
In case anyone was following along as we discussed the potential need for adding a costly migration to add an index, we ended up not needing to add one; this query is fairly quick, much quicker than the one we originally replaced, and makes use of already existing indexes.

We also only schedule jobs when we *need* to schedule jobs, so even if this was kind of slow it wouldn't matter nearly as much as it did when we originally optimized this.